hardware/tests/test_detect.py

```diff
@@ -26,23 +24,39 @@
 from hardware.tests.utils import sample
 
 
-class Keeper:
+class TestDetect(unittest.TestCase):
 
-    def __init__(self, fname, rets):
-        self.rets = rets
-        self.fname = fname
+    def setUp(self):
+        self.patch_ethtool_status = mock.patch(
+            'hardware.detect_utils.get_ethtool_status',
+            return_value=[]
+        )
+        self.patch_ethtool_status.start()
```

Comment:
Let's not use mock.start approach. First of all, if you do start, you must do stop somewhere. There are, however, less annoying ways to do it, e.g. with a class-level decorator and a complete replacement, like:
```
@mock.patch('hardware.detect_utils.get_ethtool_status', lambda *args, **kwargs: [])
class TestDetect(unittest.TestCase):
```

Reply:
I will use a different approach as suggested.
I was indeed stopping all at the end of setUp using addCleanup.